swc/private/swc.bzl

```diff
@@ -21,60 +22,89 @@ _outputs = {
 }
 
 def _impl(ctx):
-    outputs = ctx.outputs.js_outs + ctx.outputs.map_outs
+    outputs = []
     source_maps = len(ctx.outputs.map_outs) > 0
+    binding = ctx.toolchains["@aspect_rules_swc//swc:toolchain_type"].swcinfo.binding
+    args = ctx.actions.args()
 
-    for src in ctx.files.srcs:
-        js_out = ctx.actions.declare_file(paths.replace_extension(src.basename, ".js"), sibling = src)
-        outs = [js_out]
-        if source_maps:
-            outs.append(ctx.actions.declare_file(paths.replace_extension(src.basename, ".js.map"), sibling = src))
-        args = ctx.actions.args()
-
-        # Add user specified arguments *before* rule supplied arguments
-        args.add_all(ctx.attr.args)
-
-        # Pass in the swcrc config if it is set; if it is
-        # then source paths are expected to be relative to the swcrc directory
-        src_path = src.path
-        if ctx.file.swcrc:
-            swcrc_path = ctx.file.swcrc.path
-            swcrc_directory = paths.dirname(swcrc_path)
-            args.add_all([
-                "--config-file",
-                swcrc_path,
-            ])
-
-            # TODO(greg): determine if this is needed, given that relativize below should provide a correct path
-            # if not src_path.startswith(swcrc_directory):
-            #     fail("sources must be in swcrc directory or subdirectory if swcrc is specified")
-            src_path = paths.relativize(src_path, swcrc_directory)
+    # Add user specified arguments *before* rule supplied arguments
+    args.add_all(ctx.attr.args)
 
+    if ctx.attr.output_dir:
+        if len(ctx.attr.srcs) != 1:
+            fail("Under output_dir, there must be a single entry in srcs")
+        if not ctx.files.srcs[0].is_directory:
+            fail("Under output_dir, the srcs must be directories, not files")
+        out = ctx.actions.declare_file(ctx.label.name)
+        outputs.append(out)
         args.add_all([
-            src_path,
-            "--out-file",
-            js_out.path,
+            ctx.files.srcs[0].path,
+            "--out-dir",
+            out.path,
             "--no-swcrc",
             "-q",
```

Comment:
suppose we should make `-q` configurable?

Reply:
I'm not sure, I guess we'll see if someone wants that. We could send that output to a .log file instead of the bazel stdout